source/extensions/filters/http/grpc_json_transcoder/json_transcoder_filter.cc

```diff
@@ -107,114 +118,187 @@ class TranscoderImpl : public Transcoder {
 
 } // namespace
 
-JsonTranscoderConfig::JsonTranscoderConfig(
-    const envoy::extensions::filters::http::grpc_json_transcoder::v3::GrpcJsonTranscoder&
-        proto_config,
-    Api::Api& api) {
+void AsyncReflectionFetcher::requestFileDescriptors(DescriptorPoolBuilder* descriptor_pool_builder) {
+  descriptor_pool_builder_ = descriptor_pool_builder;
+  // Register to init_manager, force the listener to wait for completion of the
+  // reflection Rpcs.
+  init_target_ = std::make_unique<Init::TargetImpl>("JsonGrpcFilter: Grpc Reflection Rpcs",
+      [this]() -> void { startReflectionRpcs(); });
+  init_manager_.add(*init_target_);
+}
 
-  disabled_ = proto_config.services().empty();
-  if (disabled_) {
-    return;
+void AsyncReflectionFetcher::startReflectionRpcs() {
+  envoy::config::core::v3::GrpcService config;
+  config.mutable_envoy_grpc()->set_cluster_name(cluster_name_);
+
+  for (std::string service: services_) {
+    std::unique_ptr<Envoy::Grpc::RawAsyncClient> raw_async_client =
+      std::make_unique<Envoy::Grpc::AsyncClientImpl>(cluster_manager_, config, time_source_);
+    Envoy::Grpc::AsyncClient<ServerReflectionRequest, ServerReflectionResponse>* async_client =
+      new Envoy::Grpc::AsyncClient<ServerReflectionRequest, ServerReflectionResponse>(std::move(raw_async_client));
+    async_clients_.emplace_back(async_client);
+
+
+    Http::AsyncClient::StreamOptions options;
+    // TODO(hqin): Make this timeout part of the configuration
+    options.setTimeout(std::chrono::milliseconds(5000));
```

Comment:
good idea,  create a ReflectionConfig message in the config proto, put timeout, cluster_name to it.